### PR TITLE
Basic file loading service

### LINKED ISSUE #24: Filesystem Handling
Currently, launch issues can arise due to not having normalized paths for files that are used as input to the engine (i.e. the Basic and Texture shaders). 

As per Ruby's suggestion, this could be resolved by a filesystem handling service or something of that nature, that would allow the engine core to properly find and open the files being called.
(Input is appreciated on this)

### LINKED ISSUE #79: [Proposal] Asset Database
Currently, we have no implementation for an asset loading/registering system. This is useful for many different things, and also relates to #24 - an asset database would be nice for loading and resolving of types.

Example: (C# due to my inexperience with C++)
```csharp
var brickTexture = await AssetDatabase["textures"].GetAsync("brick.png");
```

We could also implement special folder handling. Like the above example, to register a special folder for "textures" it could be:
```csharp
AssetDatabase.RegisterSpecialFolder("textures", "./Textures", createFolder: true);
```

This would allow for flexibility and some abstraction for user end assets over the file system proposal.

### PR DESCRIPTION
**Pull request by rob5300**
This is to introduce the basic structure idea of the Asset loader.
I have replaced an instance of a shader load from disk with the loader showing how it would be used.
I have modified main to get the launch path, this will probably need changes to fit styling elsewhere.
this PR resolves #24

EDIT: Corrected ticket number - Ruby.

### CODE REVIEW

**Comment by RubyNova** [comment]
Modified the issue number. Reviewing now.

File: src/AssetLoader.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+        std::string LoadShader(std::string name);
+
+    protected:
+        static std::string modulePath;
+
+        std::string shaderFolder = "\\Shaders\\";
+
+    private:
+        static std::string GetDirectory(std::string path);
```
Comment (RubyNova) [inline src/AssetLoader.h]: Couldn't this be a `const` method?

Comment (RubyNova) [inline src/AssetLoader.h]: can we make this a `const std::string&` rather than just an `std::string`?

File: src/AssetLoader.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (RubyNova) [inline src/AssetLoader.h]: Please don't rely on this and use include guards.

File: src/NovelAssetLoader.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#include "NovelAssetLoader.h"
+#include <vector>
+#include <fstream>
+#include <sstream>
+
+namespace NovelRT{
```
Comment (Tylertron1998) [inline src/NovelAssetLoader.cpp]: nit: no space between name and opening brace

**Comment by RubyNova** [comment]
What is happening with this PR?

**Comment by tannergooding** [comment]
@rob5300, could you please indicate what's blocking the PR or sign the CLA so someone else can pick it up if you no longer wish to contribute it?

We plan on closing this tomorrow if we've not gotten a response so someone else can start on this a new.

**Comment by RubyNova** [comment]
CLA not signed, branch is currently over 200 commits behind and hasn't been committed to since 9th November. Closing this PR and reopening the ticket.

**Pull request closed without merging**